### ISSUE TITLE
Crash when plotting ongoing 3-axis sweep

### ISSUE THREAD

**Issue by abargerbos** [toolsforexperiments/plottr#28]
I'm trying to plot an unfinished run with three axes (say two RF powers and an RF frequency), something like 
```
sweep(station.hetsrc.RF.power,spec_powers)(sweep(station.qubsrc.RF.power,qub_powers)(
    ss.measure_soft_time_avg_spec(spec_frequencies, spec_src, setup=False, 
                                  setup_awg=False)))
```
where we have different dimensions for each of these sweeps, say 10, 20, and 50 points (not sure if that is relevant). This results in a crash with the following error:
```
python apps\inspectr.py --dbpath=d:\OneDrive\Setups\LK1\LK1_data\experiments.db
Traceback (most recent call last):
  File "d:\onedrive\setups\lk1\lk1_code\code\plottr\plottr\plot\mpl.py", line 238, in setData
    self._plot2d(data, ax, axesNames[0], axesNames[1], dn)
  File "d:\onedrive\setups\lk1\lk1_code\code\plottr\plottr\plot\mpl.py", line 197, in _plot2d
    im = ppcolormesh_from_meshgrid(ax, x, y, z)
  File "d:\onedrive\setups\lk1\lk1_code\code\plottr\plottr\plot\mpl.py", line 73, in ppcolormesh_from_meshgrid
    if np.any(np.isnan(x)) or np.any(np.isnan(y)):
TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

```

This only occurs when the measurement is still on-going, so I am guessing that it might be related to the guessing of the shape?

**Comment by wpfff** [comment]
oh -- probably related to the case when the data type is not float. should be a fairly quick fix.

**Comment by abargerbos** [comment]
I did not notice it before, but the plottr log actually also outputs something when the first trace is being measured. It says

```
2019-01-09 11:40:48 - plottr.node.dim_reducer.XYAxesSelector - ERROR
    Reduction on axis RF_power did not result in the right data shape. Expected (1, 60) but got (60,).
```

**Comment by wpfff** [comment]
i think this should be fixed with PR #39 -- waiting for confirmation, though.

**Comment by wpfff** [comment]
Hey @abargerbos, i haven't heard from this in a while, so i hope it's actually gone. let me know if you still have issues with it.
